File: day1/report-repair.go
package main

import (
	"fmt"
	"io"
	"os"
)

func get_input_nums() []int {
	file, err := os.Open("input.txt")

	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}

	var perline int
	var nums []int

	for {
		_, err := fmt.Fscanf(file, "%d\n", &perline)
		if err != nil {
			if err == io.EOF {
				break
			}
			fmt.Println(err)
			os.Exit(1)
		}
		nums = append(nums, perline)
	}
	return nums
}

func puzzle1(nums []int) int {
	var ans int
	for i := 0; i < len(nums); i++ {
		for j := i; j < len(nums); j++ {
			if nums[i]+nums[j] == 2020 {
				ans = nums[i] * nums[j]
				break
			}
		}
	}
	return ans
}

func puzzle2(nums []int) int {
	var ans int
	for i := 0; i < len(nums); i++ {
		for j := i; j < len(nums); j++ {
			for k := j; k < len(nums); k++ {
				if nums[i]+nums[j]+nums[k] == 2020 {
					ans = nums[i] * nums[j] * nums[k]
					break
				}
			}
		}
	}
	return ans
}

func main() {
	var nums []int = get_input_nums()
	fmt.Println(puzzle1(nums))
	fmt.Println(puzzle2(nums))
}

File: day01/report-repair.py
def puzzle1(fin):
    with open(fin, 'r') as file:
        l = file.read().split('\n')
        l = [int(s) for s in l if s != ""]
    for i in range(len(l)):
        for j in range(i, len(l)):
            if l[i] + l[j] == 2020:
                return l[i] * l[j]

def puzzle2(fin):
    with open(fin, 'r') as file:
        l = file.read().split('\n')
        l = [int(s) for s in l if s != ""]
    for i in range(len(l)):
        for j in range(i, len(l)):
            for k in range(j, len(l)):
                if l[i] + l[j] + l[k] == 2020:
                    return l[i] * l[j] * l[k]

if __name__ == "__main__":
    print(puzzle1("input.txt"))
    print(puzzle2("input.txt"))